### ISSUE TITLE
bug: ion-router does not work with query params

### ISSUE THREAD

**Issue by bubbleupdotnet** [ionic-team/ionic#19707]
# Bug Report

**Ionic version:**
4.9.1

**Current behavior:**
I use ion-router component in stencilJs.

Here is ion-router config:
```
<ion-router useHash={false}>
    <ion-route url="/home" component="app-home-page" />
</ion-router>
```

I need to open /home page with query params (/home?p=1)

Navigation to /home?p=1 does not work. I tried to navigate using ion-router-link component:
`<ion-router-link href={"/home?p=1"}>Home page</a>`
and using ion-router component
```
const nav = document.querySelector('ion-router');
nav.push("/home?p=1")
```
In both cases there is an error console: route not found. **But if I reload the page in browser then page is successfully opened.**

**Expected behavior:**

Navigation to /home?p=1 must work.

**Steps to reproduce:**

1. In stencilJs add ion-router:
```
<ion-router useHash={false}>
    <ion-route url="/home" component="app-home-page" />
</ion-router>
```
2. try to navigate to "/home?p=1"

**Comment by miqmago** [comment]
Same here, I have defined like this:

```tsx
    render() {
        return (
            <ion-app>
                <ion-router useHash={true}>
                    <ion-route url='/home' component='app-home' />
                    <ion-route url='/home/profile' component='app-profile' />
                    <ion-route-redirect from='/' to='/home'></ion-route-redirect>
                </ion-router>
                <ion-nav />
            </ion-app>
        );
    }
```

When loading web page with `/#/home/profile?age=21` it simply stays in `/#/home`.

**Comment by mslourens** [comment]
Any updates on this bug? It's pretty weird that something straightforward and commonly used as a querystring is not supported by the ion-router.

**Comment by tricki** [comment]
I just tested this in `@ionic/core@5.1.1` and it navigates correctly and sets the query string.

This might have been fixed with eab3373 which, according to the [changelog](https://github.com/ionic-team/ionic/blob/master/CHANGELOG.md#510-aluminum-2020-04-30) was released with 5.1.0, 

Also see #21062

@mslourens Can you test it in 5.1+?

**Comment by mslourens** [comment]
@tricki the issue has been solved in 5.1.1. I just tested it, the querystring gets passed along with the route.

**Comment by tricki** [comment]
@bubbleupdotnet Can you confirm this works and close the issue? Thanks.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This was resolved in Ionic Framework v5.1.0.
